agent/app/agent_unix.go

```diff
@@ -183,3 +187,25 @@ func contains(capabilities []string, capability string) bool {
 
 	return false
 }
+
+func (agent *ecsAgent) initializeResourceFields() {
```

Comment:
Why is this separate method needed? Why can't this be done conditionally, in `cgroupInit()`? Is it mostly for testing?

Reply:
Thats right, it is mainly for testing `doStart` where we mock cgroup control object for init failure.

Reply:
Can you please add documentation for this method explaining the same?